app/publisher/publisher.py

```diff
@@ -0,0 +1,37 @@
+import google
+from google.cloud.pubsub_v1 import PublisherClient
+from google.cloud.pubsub_v1.futures import Future
+from structlog import get_logger
+
+from app.publisher.publication_failed import PublicationFailed
+
+logger = get_logger(__name__)
+
+
+class PubSub(PublisherClient):
+    def __init__(self, topic_id):
+        super().__init__()
+
+        _, project_id = google.auth.default()
+        self.topic_path = super().topic_path(project_id, topic_id)
+
+    def _publish(self, message):
+        logger.info("publishing message")
+        publish_future: Future = super().publish(self.topic_path, message)
+
+        return publish_future
+
+    def publish_and_resolve_message(self, message: bytes):
+        publish_future = self._publish(message)
+        try:
+            # Resolve the future
+            publish_future.result()
+        except Exception as ex:  # pylint:disable=broad-except
```

Comment:
Why the broad exception here?

Reply:
Because when resolving the future, any number of exceptions can be raised. The design pattern is:
```
# Attempt to get the exception if there is one.
# If there is not one, then we know everything worked, and we can
# return an appropriate value.
```
The appropriate value in our case being the message id. If we want to know about specific errors like "TimeoutError" then that can be handled explicitly.

Reply:
Added a log message on error.